File: Documentation/5.3/Samples/csharp/Raven.Documentation.Samples/DocumentExtensions/Revisions/ClientAPI/Session/Counting.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Raven.Client;
using Raven.Client.Documents;
using Raven.Client.Json;
using Raven.Documentation.Samples.Orders;

namespace Raven.Documentation.Samples.ClientApi.Session.Revisions
{
    public class Loading
    {
        private interface IFoo
        {
            #region syntax
            long GetCountFor(string id);
            #endregion
        }

        public async Task Samples()
        {
            using (var store = new DocumentStore())
            {
                var CompanyProfile = new Company { Name = "Persian Rugs" };
                using (var session = store.OpenSession())
                {
                    #region example_sync
                    var revisionsCount = session.Advanced.Revisions.GetCountFor(CompanyProfile.Id);
                    #endregion
                }

                using (var asyncSession = store.OpenAsyncSession())
                {
                    #region example_async
                    var revisionsCount = await asyncSession.Advanced.Revisions.GetCountForAsync(CompanyProfile.Id);
                    #endregion
                }
            }
        }
    }
}

File: Documentation/5.4/Samples/python/DocumentExtensions/Revisions/ClientAPI/Session/Counting.py
from examples_base import ExampleBase


class CountingRevisions(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_count_revisions(self):
        with self.embedded_server.get_document_store("CountingRevisions") as store:
            with store.open_session() as session:
                # region getCount
                # Get the number of revisions for document 'companies/1-A'
                revisions_count = session.advanced.revisions.get_count_for("companies/1-A")
                # endregion


# region syntax
def get_count_for(self, id_: str) -> int: ...


# endregion
